pkg/connector/connector.go

```diff
@@ -43,6 +50,9 @@ type Connector interface {
 	Config() Config
 	SetConfig(Config)
 
+	ProvisionedBy() ProvisionType
+	SetProvisionedBy(ProvisionType)
```

Comment:
I wonder if we should remove this method, it implies that `ProvisionType` is a mutable field. I feel like this field shouldn't be able to change.